selftests/all/unit/avocado/tree_unittest.py

```diff
@@ -165,6 +166,21 @@ def test_advanced_yaml(self):
                          oldroot.children[1].children[2].value)
         self.assertEqual({'new_value': 'something'},
                          oldroot.children[3].children[0].children[0].value)
+        # multiplex root (always True)
+        self.assertEqual(tree2.multiplex, True)
+        # multiplex /virt/
+        self.assertEqual(tree2.children[0].multiplex, True)
+        # multiplex /virt/hw
+        self.assertEqual(tree2.children[0].children[0].multiplex, True)
+        # multiplex /virt/distro
+        self.assertEqual(tree2.children[0].children[1].multiplex, False)
+        # multiplex /virt/env
+        self.assertEqual(tree2.children[0].children[2].multiplex, False)
+        # multiplex /virt/absolutly
+        self.assertEqual(tree2.children[0].children[3].multiplex, True)
+        # multiplex /virt/distro/fedora
+        self.assertEqual(tree2.children[0].children[1].children[0].multiplex,
+                         True)
 
 
 class TestPathParent(unittest.TestCase):
```

Comment:
I don't think we have a general rule, other than keeping the code modular and split in modules and classes that have clear roles and names.

As you said, tree.py slowly grew into something that's not just a tree anymore. This review and the fact it finally got a multiplex reference inside shows it's time for a refactoring (which can be simple: like I said above, maybe it should just be renamed from core/tree.py to something else... since the multiplexer is a complex part of avocado, maybe it deserves its own directory in the source code).

The fact it's in the same repository is basically irrelevant.